packages/components/src/public/QueryModel/withQueryModels.tsx

```diff
@@ -520,7 +539,73 @@ export function withQueryModels<Props>(
             }
         };
 
-        loadQueryInfo = async (id: string, loadRows = false, loadSelections = false): Promise<void> => {
+        loadTotalCount = async (id: string, reloadTotalCount = false): Promise<void> => {
+            const currentModel = this.state.queryModels[id];
+
+            if (isLoading(currentModel.queryInfoLoadingState)) {
+                return;
+            }
+
+            // if we've already loaded the totalCount, no need to load it again
+            if (!reloadTotalCount && currentModel.totalCountLoadingState === LoadingState.LOADED) {
+                return;
+            }
+
+            // if usage didn't request loading the totalCount, skip it
+            if (!currentModel.includeTotalCount) {
+                this.setState(
+                    produce<State>(draft => {
+                        draft.queryModels[id].totalCountLoadingState = LoadingState.LOADED;
+                    })
+                );
+                return;
+            }
+
+            this.setState(
+                produce<State>(draft => {
+                    draft.queryModels[id].totalCountLoadingState = LoadingState.LOADING;
+                })
+            );
+
+            try {
+                const { rowCount } = await selectRows({
+                    ...currentModel.loadRowsConfig,
+                    sort: undefined,
+                    maxRows: 1,
+                    offset: 0,
+                    includeDetailsColumn: false,
+                    includeUpdateColumn: false,
+                    // includeMetadata: false, // TODO don't require metadata in selectRows response processing
+                    includeTotalCount: true,
+                });
+
+                this.setState(
+                    produce<State>(draft => {
+                        const model = draft.queryModels[id];
+                        model.rowCount = rowCount;
+                        model.totalCountLoadingState = LoadingState.LOADED;
+                        model.totalCountError = undefined;
+                    })
+                );
+            } catch (error) {
+                this.setState(
+                    produce<State>(draft => {
+                        const model = draft.queryModels[id];
+                        let rowsError = resolveErrorMessage(error);
+
+                        if (rowsError === undefined) {
+                            rowsError = `Error while loading total count for SchemaQuery: ${model.schemaQuery.toString()}`;
```

Comment:
If this is an error the user will see, we might not want to include the `SchemaQuery: ...` technical bit.  It's more informative (to us), but not a language they speak usually.

Reply:
Here I just copied the similar error message as set in the loadRows case for the `rowsError`. This is the fallback error. Hopefully the error returned from the server that goes through the `resolveErrorMessage` is more user friendly. I'm happy to remove the schemaQuery part from both of these cases if you think that would be better.